storage/mysql/tree_storage.go

```diff
@@ -283,6 +308,12 @@ func (t *treeTX) storeSubtrees(ctx context.Context, subtrees []*storagepb.Subtre
 	// a really large number of subtrees to store.
 	args := make([]interface{}, 0, len(subtrees))
 
+	subtreeRev := t.writeRevision
```

Comment:
Would flipping the logic here be easier for the reader?

```golang
var subtreeRev int64
if t.subtreeRevs {
  // We're using subtrees so ensure we write at the correct revision:
  subtreeRev = t.writeRev
}
```

Separately, is there anything special about the `0` value? The current comment sort of implies it's arbitrary, but e.g. does it break if the `treeTX` default of `-1` were used?

Reply:
+1 that caught my attention as well

Reply:
Will look at this again tomorrow. Today has been spent doing heavy lifting. I'll come back around for a nuance pass after some time away from this code :-)

Reply:
I've gone with Al's suggestion here. `-1` would also work, as would anything else. We just need a non-`NULL` value to write, and then it will never be read again.